### PR TITLE
[DOCS] Productionizing Guide #56

### LINKED ISSUE #56: [DOCS] Productionizing Guide
## Issue Overview

We need some guide on the documentation to suggest to users steps they could take to optimize for production build

## Proposed Solutions or Ideas

Write under the "Shipping your dApp" section.

Some potential content:
- Removal of debug page to reduce bundle size
- Image compression (if any).

### PR DESCRIPTION
**Pull request by martinvibes**
# Task name here
[DOCS] Productionizing Guide #56

Fixes #56 

## Types of change

- [] Bug
- [x] Enhancement

### CODE REVIEW

**Comment by martinvibes** [comment]
hey @metalboyrick kindly review :)

File: docs/deploying/deploy-production.mdx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+# Check versions
+node --version        # >= v18.17
+starknet-devnet --version  # 0.2.3
+scarb --version      # 2.9.2
+snforge --version    # 0.34.0
+```
+
+:::tip Hint
+Using incompatible versions may cause unexpected issues in production. Make sure to align with the compatible versions specified in Scaffold-Stark 2.
+:::tip Hint
```
Comment (metalboyrick) [inline docs/deploying/deploy-production.mdx:24]: use warning here

File: docs/deploying/deploy-production.mdx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+:::tip Hint
+Only add domains to `remotePatterns` that you trust and need for your application. For each additional pattern may increases the potential attack surface of the application.
+:::tip Hint
+
+### Additional Build optimizations
+
+Scaffold-Stark 2 includes several build optimizations in `next.config.mjs`:
+
+#### 1. TypeScript and ESLint error handling:
```
Comment (metalboyrick) [inline docs/deploying/deploy-production.mdx:96]: I think need more information here, considering many beginners would want to use the framework

File: docs/deploying/deploy-production.mdx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+```javascript
+typescript: {
+  ignoreBuildErrors: process.env.NEXT_PUBLIC_IGNORE_BUILD_ERROR === "true",
+},
+eslint: {
+  ignoreDuringBuilds: process.env.NEXT_PUBLIC_IGNORE_BUILD_ERROR === "true",
+},
+```
+
+#### 2. Webpack optimization for compatibility:
```
Comment (metalboyrick) [inline docs/deploying/deploy-production.mdx:107]: I think need more information here, considering many beginners would want to use the framework

**Review by metalboyrick — CHANGES_REQUESTED** [review]
some comments, will chime in @iossocket for review too, thanks for the PR!

**Comment by martinvibes** [comment]
hey @metalboyrick implemented requested changes :)

**Review by metalboyrick — APPROVED** [review]
looks good to me but let's chime in @iossocket for review.

**Review by Nadai2010 — APPROVED** [review]

**Comment by metalboyrick** [comment]
will be iterated within the team cc @Nadai2010 @iossocket

**Pull request merged**